src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -476,13 +476,19 @@ class Wav2Vec2FeatureProjection(nn.Module):
     def __init__(self, config):
         super().__init__()
         self.layer_norm = nn.LayerNorm(config.conv_dim[-1], eps=config.layer_norm_eps)
-        self.projection = nn.Linear(config.conv_dim[-1], config.hidden_size)
+        self.projection = None
+        if config.conv_dim[-1] != config.hidden_size:
+            self.projection = nn.Linear(config.conv_dim[-1], config.hidden_size)
         self.dropout = nn.Dropout(config.feat_proj_dropout)
 
     def forward(self, hidden_states):
         # non-projected hidden states are needed for quantization
         norm_hidden_states = self.layer_norm(hidden_states)
-        hidden_states = self.projection(norm_hidden_states)
+        hidden_states = norm_hidden_states.clone()
```

Comment:
Is this potentially a bit cleaner?
```suggestion
        if self.projection is not None:
            hidden_states = self.projection(hidden_states)
        else:
            hidden_states = norm_hidden_states.clone()
```